jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -654,32 +697,299 @@ def _set_value_in_subset_definition(self, index, name, desired_key, new_value):
+                        bad_region.append((region, 'Sky region provided but data has no valid WCS'))  # noqa
+                        continue
+                    region = region.to_pixel(data.coords)
+
+                if hasattr(region, 'to_mask'):
+                    try:
+                        mask = region.to_mask(**kwargs)
+                        im = mask.to_image(data.shape)  # Can be None
+                    except Exception as e:  # pragma: no cover
+                        bad_regions.append((region, f'Failed to load: {repr(e)}'))
+                        continue
+
+                # Boolean mask as input is supported but not advertised.
+                elif (isinstance(region, np.ndarray) and region.shape == data.shape
+                      and region.dtype == np.bool_):
+                    im = region
+
+                if im is None:
+                    bad_regions.append((region, 'Mask creation failed'))
+                    continue
+
+                # NOTE: Region creation info is thus lost.
+                try:
+                    subset_label = f'{msg_prefix} {msg_count}'
+                    state = MaskSubsetState(im, data.pixel_component_ids)
+                    self.app.data_collection.new_subset_group(subset_label, state)
+                    msg_count += 1
+                except Exception as e:  # pragma: no cover
+                    bad_regions.append((region, f'Failed to load: {repr(e)}'))
+                    continue
             else:
-                m = mode
-            self.app.session.edit_subset_mode.mode = m
-            s = RangeSubsetState(lo=sub_region.lower.value, hi=sub_region.upper.value,
-                                 att=range_viewer.slice_component_label)
-            range_viewer.apply_subset_state(s)
+                bad_regions.append((region, 'Mask creation failed'))
+                continue
+            n_loaded += 1
+            if max_num_regions is not None and n_loaded >= max_num_regions:
+                break
+
+        n_reg_in = len(regions)
+        n_reg_bad = len(bad_regions)
+        if n_loaded == 0:
+            snack_color = "error"
+        elif n_reg_bad > 0:
+            snack_color = "warning"
+        else:
+            snack_color = "success"
+        self.app.hub.broadcast(SnackbarMessage(
+            f"Loaded {n_loaded}/{n_reg_in} regions, max_num_regions={max_num_regions}, "
+            f"bad={n_reg_bad}", color=snack_color, timeout=8000, sender=self.app))
+
+        if return_bad_regions:
+            return bad_regions
+
+    @observe('combination_selected')
+    def _combination_selected_updated(self, change):
+        self.app.session.edit_subset_mode.mode = SUBSET_MODES_PRETTY[change['new']]
+
+    def _update_combination_mode(self):
+        for key, value in SUBSET_MODES_PRETTY.items():
+            if self.app.session.edit_subset_mode.mode == value:
+                self.combination_mode.selected = key
+                return
```

Comment:
if this would ever need to be done in more than one place, we could have an inverted dictionary defined at the top (although then the dictionaries should probably be named something that makes it clear which is which direction).

```
SUBSET_TO_PRETTY = {v: k for k,v in SUBSET_MODES_PRETTY.items()}
```